eox_nelp/signals/receivers.py

```diff
@@ -16,10 +18,11 @@ def block_completion_progress_publisher(instance, **kwargs):  # pylint: disable=
     Args:
         instance<Blockcompletion>: Instance of BlockCompletion model.
     """
-    dispatch_futurex_progress.delay(
-        course_id=str(instance.context_key),
-        user_id=instance.user_id,
-    )
+    if getattr(settings, "ACTIVATE_DISPATCH_FUTUREX_PROGRESS", False):
```

Comment:
why not inside dispatch_futurex_progress task ?

Reply:
is a good question, probably with this approach you avoid the generation of the task in celery if is not activated(block completion case). The question is which is better to ask for setting inside the celery task or not?